swagger.yaml

```diff
@@ -1,42 +1,376 @@
+      - FlorenceAPIKey: []
+      responses:
+        200:
+          description: "The image publishing was successfully requested to Static file publisher."
+        400:
+          description: "Invalid request, image id was incorrect"
+        401:
+          $ref: '#/responses/Unauthenticated'
+        403:
+          description: "Unauthorised to publish image"
+        404:
+          $ref: '#/responses/NotFound'
+        409:
+          description: "Cannot publish the image because it is not uploaded yet, or it is has already been published"
         500:
           $ref: '#/responses/InternalError'
 
 responses:
+
   InternalError:
     description: "Failed to process the request due to an internal error"
+    
+  NotFound:
+    description: "Requested item cannot be found"
+    
+  InvalidRequestError:
+    description: "Failed to process the request due to invalid request"
+
+  Unauthenticated:
+    description: "User or service is not authenticated"
 
 definitions:
-  HelloResponse:
+
+  Images:
+    description: "A list of images"
+    type: object
+    properties:
+      count:
+        description: "The number of images returned"
+        readOnly: true
+        type: integer
+        example: 1
+      items:
+        type: array
+        items:
+          $ref: '#/definitions/Image'
+      limit:
+        description: "The number of images requested"
+        type: integer
+      offset:
+        description: "The first row of images to retrieve, starting at 0. Use this parameter as a pagination mechanism along with the limit parameter"
+        type: integer
+      total_count:
+        description: "The total number of images"
+        readOnly: true
+        type: integer
+        example: 1
+
+  NewImage:
+    type: object
+    description: "New Image metadata model, used to create new images. The image ID will be created by image API, and the state will be set to 'created' at creation time."
+    properties:
+      collection_id:
+        type: string
+        description: "Collection unique identifier corresponding to this image"
+        example: "5557dcd9-bf58-4a67-94f7-2343569834cc"
+      filename:
+        type: string
+        description: "Image's file name. According to SEO recommendations, the name should not be longer than 5 words. And it should not include extension because multiple extensions for the same image might be available as download variants."
+        example: "image-name"
+        maxLength: 40
+      license:
+        type: object
+        description: "License that defines the third party permissions for this image"
+        properties:
+          title:
+            type: string
+            description: "Title of the license"
+            example: "Open Government Licence v3.0"
+          href: 
+            type: string
+            description: "Link to the license content"
+            example: "https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
+      upload:
```

Comment:
As the image resource is created before the upload (as we need the ID to be generated first), we will not have the `upload` yet in the new image resource so it can be removed here.